device/devices.go

```diff
@@ -0,0 +1,52 @@
+package device
+
+import (
+	"log"
+	"net"
+	"os/exec"
+	"regexp"
+	"time"
+)
+
+type Dev struct {
+	// houses all device information
+	DeviceID         int
+	CommonName       string
+	Name             string
+	Description      string
+	Addresses        []Address
+	TimeRegistered   time.Time
+	TimeDeregistered time.Time
+	PacketCount      int
+	DeviceSniffs     []SniffProcess
+}
+type Address struct {
+	// Used to house addresses of devices
+	IP     net.IP
+	Subnet net.IPMask
+}
+
+func GetNetworkDeviceInfo() []Dev {
+	// uses t-hark to find sniff-able devices
+	var devInfo []Dev
+	var deviceFormat = regexp.MustCompile(`(\d+.\s)`)
+	var interfacePatt = regexp.MustCompile(`(?:\w+)`)
+	out, err := exec.Command("tshark", "-D").Output()
+
+	sniffableDevices := deviceFormat.Split(string(out), -1)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
does the caller not care it might never get this back? I try to avoid panics, favoring returning an error with some additional context unless I'm the main package, and then sometimes I'll allow it (but not always)